ebean-datasource/src/main/java/io/ebean/datasource/pool/ConnectionPool.java

```diff
@@ -377,6 +377,9 @@ private void testConnection() {
     } finally {
       try {
         if (conn != null) {
+          if (!conn.getAutoCommit()) {
+            conn.rollback();
```

Comment:
This fixes the connection leak with DB2.
When we just close the connection (=return to pool) and there is an open UOW, the pool cannot close the connection on trim and the pool restarts. This connection stays open and also their tcp/ip connection (more detailed explanation in #107)